libs/template/file.go

```diff
@@ -97,5 +121,36 @@ func (f *inMemoryFile) PersistToDisk() error {
 	if err != nil {
 		return err
 	}
-	return os.WriteFile(path, f.content, f.perm)
+
+	ctx := context.Background()
+
+	if runsOnDatabricks(ctx) && f.isNotebook() {
+		return writeNotebook(ctx, path, f.content)
+	} else {
+		return os.WriteFile(path, f.content, f.perm)
+	}
+}
+
+func runsOnDatabricks(ctx context.Context) bool {
+	_, ok := env.Lookup(ctx, "DATABRICKS_RUNTIME_VERSION")
```

Comment:
We should really do this in just one way throughout the codebase. What we discussed is checking that env var + the existence of /Workspace.

Reply:
extracted this code into a shared `runtime` lib